common/collection/sync_map.go

```diff
@@ -68,3 +86,13 @@ func (m *SyncMap[K, V]) Pop(key K) (value V, ok bool) {
 	}
 	return value, ok
 }
+
+func (m *SyncMap[K, V]) KeySet() []K {
+	m.Lock()
+	defer m.Unlock()
+	keys := make([]K, 0, len(m.contents))
+	for k := range m.contents {
+		keys = append(keys, k)
+	}
+	return keys
```

Comment:
```suggestion
	return maps.Keys(m.contents)
```
?

(using `"golang.org/x/exp/maps"`, not the stdlib)